src/components/tx-flow/flows/SignMessage/SignMessage.tsx

```diff
@@ -212,54 +211,48 @@ const MsgModal = ({
-          <DialogTitle onClose={handleClose} logoUri={logoUri} name={name} />
-
-          <DialogContent>
-            <DialogHeader threshold={safe.threshold} />
-
-            <Typography fontWeight={700} mb={1}>
-              Message: <CopyButton text={decodedMessageAsString} />
-            </Typography>
-            <DecodedMsg message={decodedMessage} isInModal />
-
-            <MessageHashField label="SafeMessage" hashValue={safeMessageMessage} />
-            <MessageHashField label="SafeMessage hash" hashValue={safeMessageHash} />
-
-            <AlreadySignedByOwnerMessage hasSigned={hasSigned} />
-
-            <InfoBox
-              message={
-                requestId
-                  ? 'Please keep this modal open until all signers confirm this message. Closing the modal will abort the signing request.'
-                  : 'The signature will be submitted to the Safe App when the message is fully signed.'
-              }
-            >
-              <MsgSigners
-                msg={ongoingMessage || createSkeletonMessage(safe.threshold)}
-                showOnlyConfirmations
-                showMissingSignatures
-                backgroundColor={palette.info.background}
-              />
-            </InfoBox>
-
-            <WrongChainWarning />
-
-            <MessageDialogError isOwner={isOwner} submitError={submitError} />
-          </DialogContent>
-
-          <DialogActions>
-            <Button onClick={handleClose}>Cancel</Button>
-            <Button variant="contained" color="primary" onClick={onSign} disabled={isDisabled}>
-              Sign
-            </Button>
-          </DialogActions>
-        </div>
-      </ModalDialog>
+      <DialogContent>
+        <DialogHeader threshold={safe.threshold} />
+
+        <Typography fontWeight={700} mb={1}>
+          Message: <CopyButton text={decodedMessageAsString} />
+        </Typography>
+        <DecodedMsg message={decodedMessage} isInModal />
+
+        <MessageHashField label="SafeMessage" hashValue={safeMessageMessage} />
+        <MessageHashField label="SafeMessage hash" hashValue={safeMessageHash} />
+
+        <AlreadySignedByOwnerMessage hasSigned={hasSigned} />
+
+        <InfoBox
+          message={
+            requestId
+              ? 'Please keep this modal open until all signers confirm this message. Closing the modal will abort the signing request.'
+              : 'The signature will be submitted to the Safe App when the message is fully signed.'
+          }
+        >
+          <MsgSigners
+            msg={ongoingMessage || createSkeletonMessage(safe.threshold)}
+            showOnlyConfirmations
+            showMissingSignatures
+            backgroundColor={palette.info.background}
+          />
+        </InfoBox>
+
+        <WrongChainWarning />
+
+        <MessageDialogError isOwner={isOwner} submitError={submitError} />
+      </DialogContent>
+
+      <DialogActions>
+        <Button onClick={handleClose}>Cancel</Button>
```

Comment:
Should we move this logic to the cross in the top right?

Reply:
Good idea! I will check to see if its possible to move this to the onClose call of the modal provider.

Reply:
After discussing this together we decided to solve this in a separate PR because it requires some work done on the modal provider. Essentially, we need to find a way so that child components can hook into the onClose logic of the provider to prevent it from closing. Currently, its possible to pass an onClose handler to the provider but it has to be defined outside the child components. cc @katspaugh